CRM/Core/BAO/CustomField.php

```diff
@@ -2043,6 +2043,9 @@ public static function moveField($fieldID, $newGroupID) {
     $add->custom_group_id = $newGroup->id;
     self::createField($add, 'add');
 
+    // IS NOT NULL needed here to prevent NULL values from being inserted when a field is moving from one entity type to another.
+    // It will pass _moveFieldValidate if all values are NULL,
+    // but will break things to try to insert NULL values for entity ids that make no sense for the new entity type.
```

Comment:
Non-blocking suggestion:
```suggestion
    // @todo If the entity type is different, it doesn't even make sense to copy anything since the entity id is not the correct id.
```

Reply:
I didn't mean that as a replacement text. Let me fix...

Reply:
Well, as it stands now that is what happens - we can only copy to a different entity type if there are no non-null values and we can only copy non-null values, so we can't copy values at all. So does it make sense to add a todo? Might confuse things because future readers will think it means we can copy values with different entity types, when we cannot.

Reply:
Copy, over and out.